### PR TITLE
Allow AWS Operator RedshiftToS3Transfer To Run a Custom Query

### PR DESCRIPTION
**Pull request by AratiNagmal**
**RedshiftToS3Transfer operator to run custom select query on table/s to transfer data to S3**

When a "select_query" parameter is passed to RedshiftToS3Transfer operator then it will use it, otherwise it will use the default behaviour.

Related Issue: [AIRFLOW-10124](https://github.com/apache/airflow/issues/10124)

### CODE REVIEW

File: airflow/providers/amazon/aws/transfers/redshift_to_s3.py [hunk 1]
```diff
@@ -124,10 +138,8 @@ def execute(self, context) -> None:
         credentials = s3_hook.get_credentials()
         credentials_block = build_credentials_block(credentials)
         unload_options = '\n\t\t\t'.join(self.unload_options)
-        s3_key = f"{self.s3_key}/{self.table}_" if self.table_as_file_name else self.s3_key
-        select_query = f"SELECT * FROM {self.schema}.{self.table}"
 
-        unload_query = self._build_unload_query(credentials_block, select_query, s3_key, unload_options)
+        unload_query = self._build_unload_query(credentials_block, self._select_query, self.s3_key, unload_options)
```
Comment (kaxil) [inline airflow/providers/amazon/aws/transfers/redshift_to_s3.py]: This should fix the failing static check - https://github.com/apache/airflow/pull/14177/checks?check_run_id=1873280398#step:7:213

```suggestion
        unload_query = self._build_unload_query(
            credentials_block, self._select_query, self.s3_key, unload_options
        )
```

File: airflow/providers/amazon/aws/transfers/redshift_to_s3.py [hunk 2]
```diff
@@ -101,6 +107,14 @@ def __init__(  # pylint: disable=too-many-arguments
         self.include_header = include_header
         self.table_as_file_name = table_as_file_name
 
+        self._select_query = None
+        if select_query:
+            self._select_query = select_query
+        elif self.schema and self.table:
+            self._select_query = f"SELECT * FROM {self.schema}.{self.table}"
+        else:
+            raise ValueError('Please provide both `schema` and `table` params or `select_query` to fetch the data.')
```
Comment (kaxil) [inline airflow/providers/amazon/aws/transfers/redshift_to_s3.py]:
```suggestion
            raise ValueError(
                'Please provide both `schema` and `table` params or `select_query` to fetch the data.'
            )
```

Comment (kaxil) [reply airflow/providers/amazon/aws/transfers/redshift_to_s3.py]: This should fix failing Pylint error: https://github.com/apache/airflow/pull/14177/checks?check_run_id=1873280195#step:7:137

Comment (kaxil) [reply airflow/providers/amazon/aws/transfers/redshift_to_s3.py]: Would recommend installing pre-commit to identify this before running `git commit`:

More details: https://github.com/apache/airflow/blob/master/STATIC_CODE_CHECKS.rst

Comment (AratiNagmal) [reply airflow/providers/amazon/aws/transfers/redshift_to_s3.py]: Thank you for pointers @kaxil !

**Review by kaxil — APPROVED** [review]

**Comment by kaxil** [comment]
Well done @AratiNagmal 👏

**Pull request merged**